mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -52,6 +52,61 @@ template <typename Ty> static Ty getMemRefElementType(Type ty) {
   return dyn_cast<Ty>(mt.getElementType());
 }
 
+/// Test \p id is default constructed and it is not modified.
+static bool isDefaultConstructedID(Value id) {
+  // Check it is default constructed
+  auto constructor = id.getDefiningOp<sycl::SYCLIDConstructorOp>();
+  if (!constructor || !constructor.getArgs().empty())
+    return false;
+
+  // Conservatively check it has not been modified after construction
```

Comment:
```suggestion
  // Conservatively check it has not been modified or address token after construction
```
?

Reply:
I'd rather keep the current comment. As we do not have analysis, checking the value is not being directly modified or a pointer to it is being obtained through other operation is "*convervatively* checking it has not been modified" IMO

Reply:
ok.